graph/src/runtime/value.rs

```diff
@@ -515,6 +515,124 @@ impl Value {
         }
     }
 
+    /// Convert Value to JSON string representation
+    pub fn to_json_string(
+        &self,
+        runtime: &crate::runtime::runtime::Runtime,
+    ) -> String {
+        match self {
+            Self::Null => String::from("null"),
+            Self::Bool(b) => b.to_string(),
+            Self::Int(i) => i.to_string(),
+            Self::Float(f) => {
+                // Handle special float values
+                if f.is_nan() || f.is_infinite() {
+                    String::from("null")
+                } else {
+                    // Use default float formatting
+                    f.to_string()
+                }
+            }
+            Self::String(s) => escape_json_string(s),
+            Self::List(list) => {
+                let items: Vec<String> = list.iter().map(|v| v.to_json_string(runtime)).collect();
```

Comment:
use itertools to join without collect

Reply:
with the DisplayJson approach this code is not more exists, it is write the data to the buffer as it iterate on the values